packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -1,21 +1,148 @@
+  }
+
+  /**
+   * A function to validate, construct and store loadOption, lifecycle, source and destination
+   * config related information in global state
+   */
   init() {
-    // TODO: get writekey, dataplaneUrl and loadOptions from state to use
+    validateLoadArgs(state.lifecycle.writeKey.value, state.lifecycle.dataPlaneUrl.value);
+    // determine the sourceConfig url
+    const intgCdnUrl = getIntegrationsCDNPath(
+      'process.package_version',
+      state.loadOptions.value.lockIntegrationsVersion as boolean,
+      state.loadOptions.value.destSDKBaseURL,
+    );
+
+    // determine if the staging SDK is being used
+    // TODO: deprecate this in new version and stop adding '-staging' in filenames
+    const { isStaging } = getSDKUrlInfo();
+
+    // set the final load option in state
+    batch(() => {
+      // set application lifecycle state in state
+      if (state.loadOptions.value.logLevel) {
+        state.lifecycle.logLevel.value = state.loadOptions.value.logLevel;
+      }
+      state.lifecycle.integrationsCDNPath.value = intgCdnUrl;
+      if (state.loadOptions.value.configUrl) {
+        state.lifecycle.sourceConfigUrl.value = `${state.loadOptions.value.configUrl}/sourceConfig/?p=process.module_type&v=process.package_version&writeKey=${state.lifecycle.writeKey.value}&lockIntegrationsVersion=${state.loadOptions.value.lockIntegrationsVersion}`;
+      }
+      state.lifecycle.isStaging.value = isStaging;
+    });
     this.getConfig();
   }
 
-  getConfig() {
-    // Once done call all ready
-    this.onReady();
+  /**
+   * A callback function that is executed once we fetch the source config response.
+   * Use to construct and store information that are dependent on the sourceConfig.
+   * @param res source config response
+   */
+  processConfig(res?: SourceConfigResponse | string) {
+    if (!res || typeof res !== 'object') {
+      throw Error('Unable to fetch source config');
+    }
+    // determine the dataPlane url
+    const dataPlaneUrl = resolveDataPlaneUrl(
+      res.source.dataplanes,
+      state.lifecycle.dataPlaneUrl.value,
+      state.loadOptions.value.residencyServer,
+    );
+    const nativeDestinations: Destination[] =
+      res.source.destinations.length > 0 ? filterEnabledDestination(res.source.destinations) : [];
+
+    // set in the state --> source, destination, lifecycle, reporting
+    batch(() => {
+      // set source related information in state
+      state.source.value = {
+        id: res.source.id,
+      };
+      // set device mode destination related information in state
+      state.destinations.value = nativeDestinations;
+
+      // set application lifecycle state
+      state.lifecycle.activeDataplaneUrl.value = dataPlaneUrl;
+      state.lifecycle.status.value = 'configured';
+
+      // set the values in state for reporting slice
+      state.reporting.isErrorReportingEnabled.value =
+        res.source.config.statsCollection.errorReports.enabled || false;
```

Comment:
The key path is now `statsCollection.errors.enabled`.

Reply:
This would also work in our advantage where the impact radius would be minimal in case something crazy happens after the release.
Moreover, once we pick the customers who are our initial test users, we can ask them to upgrade to the version with Bugsnag feature.